src/AssetRegistry/MainRegistry.sol

```diff
@@ -87,7 +87,9 @@ contract MainRegistry is Ownable {
         //Main registry must be initialised with usd
         baseCurrencyToInformation[baseCurrencyCounter] = baseCurrencyInformation;
         assetToBaseCurrency[baseCurrencyInformation.assetAddress] = baseCurrencyCounter;
+        isBaseCurrency[baseCurrencyInformation.assetAddress] = true;
```

Comment:
this could be a `addBaseCurrency(..)` if we change the struct calldata to memory

Reply:
Add check in testSuccess_deployment()

Reply:
done, quite similar as the testSuccess_initialize